### PR TITLE
MP-6953 [BS-8103] NOX "HashedText" - Define and implement custom type

### PR DESCRIPTION
**Pull request by becicajla**

### CODE REVIEW

**Review by rochar — DISMISSED** [review]
looks good

File: src/Nox.Types/Types/HashedText/TypeOptions/HashedTextTypeOptions.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿namespace Nox.Types;
+
+public class HashedTextTypeOptions
+{
+    public HashingAlgorithm HashingAlgorithm { get; set; } = HashingAlgorithm.SHA256;
+    public int Salt { get; set; } = 64;
```
Comment (gjorgji-grgovski-authoritypartners) [inline src/Nox.Types/Types/HashedText/TypeOptions/HashedTextTypeOptions.cs:6]: Can we call this property "SaltLength" for clarity?

File: src/Nox.Types/Types/HashedText/HashedText.cs [hunk 2]
```diff
@@ -1,9 +1,113 @@
...
+
+    /// <summary>
+    /// Creates salt byte array with length byteCount
+    /// </summary>
+    /// <param name="byteCount">array length</param>
+    /// <returns></returns>
+    private static byte[] GetSalt(int byteCount)
+    {
+        byte[] salt = new byte[byteCount];
+        RNGCryptoServiceProvider rng = new();
```
Comment (gjorgji-grgovski-authoritypartners) [inline src/Nox.Types/Types/HashedText/HashedText.cs:92]: RNGCryptoServiceProvider is obsoleted

https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.rngcryptoserviceprovider?view=net-7.0#definition

File: src/Nox.Types/Types/HashedText/HashedText.cs [hunk 3]
```diff
@@ -1,9 +1,113 @@
...
+    }
+    public string Salt
+    {
+        get => Value.Salt;
+        private set => Value = (Value.HashText, Salt: value);
+    }
+
+    public HashedText() { Value = (string.Empty, string.Empty); }
+
+    public static HashedText From(string value, HashedTextTypeOptions options)
```
Comment (gjorgji-grgovski-authoritypartners) [inline src/Nox.Types/Types/HashedText/HashedText.cs:25]: Can we have XML documentation on the public methods at least?

File: tests/Nox.Types.Tests/Types/HashedText/HashedTextTests.cs [hunk 4]
```diff
@@ -1,11 +1,79 @@
...
+
+    [Fact]
+    public void HashedText_Constructor_WithOptions_ReturnsHashedValue()
+    {
+        string text = "Text to hash";
+        byte[] textData = System.Text.Encoding.UTF8.GetBytes(text);
+        byte[] hash = SHA512.HashData(textData);
+        var textHashedExpected = Convert.ToBase64String(hash);
+
+        var hashedText = HashedText.From(text, new HashedTextTypeOptions() { HashingAlgorithm = HashingAlgorithm.SHA512, Salt = 0 });
```
Comment (gjorgji-grgovski-authoritypartners) [inline tests/Nox.Types.Tests/Types/HashedText/HashedTextTests.cs:29]: Just for readability can you move this line under "string text = "Text to hash""?
Same for the other methods below.

**Review by gjorgji-grgovski-authoritypartners — CHANGES_REQUESTED** [review]

File: src/Nox.Types/Types/HashedText/HashedText.cs (hunk 3 above)
Comment (danield-ap) [reply src/Nox.Types/Types/HashedText/HashedText.cs:42]: Documentation added.

File: src/Nox.Types/Types/HashedText/HashedText.cs (hunk 2 above)
Comment (danield-ap) [reply src/Nox.Types/Types/HashedText/HashedText.cs]: `RNGCryptoServiceProvider` replaced for `RandomNumberGenerator`.

File: src/Nox.Types/Types/HashedText/TypeOptions/HashedTextTypeOptions.cs (hunk 1 above)
Comment (danield-ap) [reply src/Nox.Types/Types/HashedText/TypeOptions/HashedTextTypeOptions.cs]: Property name updated.

**Review by gjorgji-grgovski-authoritypartners — APPROVED** [review]

**Review by rochar — CHANGES_REQUESTED** [review]
Looks good

Just need to use the TypeOptions in Nox.Solution and regenerate the schema

**Pull request merged**